### PR TITLE
NMS-10654: When editing a surveillance category from Admin flow, lists of nodes are not sorted by node label (reopened)

### PR DESCRIPTION
**Pull request by jeffgdotorg**
* JIRA: http://issues.opennms.org/browse/NMS-10654

### CODE REVIEW

**Comment by jeffgdotorg** [comment]
@RangerRick if you don't mind having a quick look at this, I would appreciate it. I used the ES6-specific construct `for (const foo of bar)` and would like to be sure that's unlikely to cause issues.

**Comment by RangerRick** [comment]
> @RangerRick if you don't mind having a quick look at this, I would appreciate it. I used the ES6-specific construct `for (const foo of bar)` and would like to be sure that's unlikely to cause issues.

IE doesn't support it at all, so depends on whether you care about that.  ;)  (same with `const` for that matter...)

In our other javascript stuff it goes through a babel compile stage so it gets turned into ES5, but if you're doing it straight in a JSP I'd still do it the old way.

File: opennms-webapp/src/main/webapp/WEB-INF/jsp/admin/editCategory.jsp [hunk 1]
```diff
@@ -73,10 +73,14 @@ function populateOptGroupFromList(elementName, list) {
 		optgroupElem.remove(0);
 	}
 	
-	for (var nodeId in list) {
+	for (const nodeSpec of list) {
```
Comment (RangerRick) [inline opennms-webapp/src/main/webapp/WEB-INF/jsp/admin/editCategory.jsp]: The (somewhat) efficient way to do this in es5 is:

```
for (var i=0, len=list.length, nodeSpec; i < len; i++) {
    nodeSpec = list[i];
```

Comment (jeffgdotorg) [reply opennms-webapp/src/main/webapp/WEB-INF/jsp/admin/editCategory.jsp]: Thanks. Glad I asked :) I'll rework it to do this instead.

**Pull request merged**